src/metadata-parser.h

```diff
@@ -29,19 +28,24 @@ namespace librealsense
     {
     public:
         md_constant_parser(rs2_frame_metadata_value type) : _type(type) {}
-        rs2_metadata_type get(const frame& frm) const override
+
+        bool find( const frame & frm, rs2_metadata_type * p_value ) const override
         {
-            rs2_metadata_type v;
-            if (try_get(frm, v) == false)
+            const uint8_t * pos = frm.additional_data.metadata_blob.data();
+            while( pos <= frm.additional_data.metadata_blob.data() + frm.additional_data.metadata_blob.size() )
             {
-                throw invalid_value_exception("Frame does not support this type of metadata");
+                const rs2_frame_metadata_value * type = reinterpret_cast<const rs2_frame_metadata_value *>(pos);
+                pos += sizeof( rs2_frame_metadata_value );
+                if( _type == *type )
+                {
+                    const rs2_metadata_type * value = reinterpret_cast<const rs2_metadata_type *>(pos);
+                    if( p_value )
```

Comment:
p_value is nullptr we return true?
Should be false, right?

Reply:
We return true if the value exists, i.e. we "support" this metadata. Has nothing to do with the value of `p_value`. `p_value` is the address where want to put the value of the metadata. If it's null, we're not interested in the value - just if it's there.